### ISSUE TITLE
Bug on wavform

### ISSUE THREAD

**Issue by luca2125** [MikeG621/YOGEME#49]
Hi Mike,

There is a bug on wav form that crash if the mission or battle  is > 9:

```
			_frontend = _wave + "FrontEnd\\" + Path.GetFileNameWithoutExtension(_mission.MissionFileName).Substring(1).Remove(4) + "\\";
			_battleNumber = int.Parse(Directory.GetParent(_frontend).Name.Substring(1, 1));
			_missionNumber = int.Parse(Directory.GetParent(_frontend).Name.Substring(3, 1));
```

You work on fixed chars: Remove(4) and when you convert to int if the mission is > 9 you get a letter and this cause exception.

So if you do something like this for example::

```
FileNoChars = Path.GetFileNameWithoutExtension(_mission.MissionFileName).Replace("B", " ").Replace("M"," ");

_battleNumber =FileNoChars.Split(FileNoChars, " ")[1];
_missionNumber =FileNoChars.Split(FileNoChars, " ")[2];
```

This is only a pseudo code to explain the concept.

Hope this help.

**Comment by MikeG621** [comment]
I knew about this when I did it, hence the comment right above that talking
about double and triple digits since what you're doing with 14+ battles
isn't normal.  It's more complicated than your pseudo-code; that method by
itself can't work, I will likely use Regex to parse it out properly.
--

**Comment by MikeG621** [comment]
Fixed in v1.9.2
